modules/effects/src/effects_module.ts

```diff
@@ -68,3 +97,22 @@ export function _provideForRootGuard(runner: EffectsRunner): any {
   }
   return 'guarded';
 }
+
+export function _createFeatureEffects(
+  injector: Injector,
+  effectCollections: Type<any>[][] | InjectionToken<Type<any>[]>[]
+): any[] {
+  const effects: Type<any>[] = [];
+
+  effectCollections.forEach(
```

Comment:
Imho it's more readable if we'd use a `map` here - see https://github.com/ngrx/platform/blob/master/modules/store/src/store_module.ts#L278 for an example

Reply:
> I'm not sure which use cases this feature has, in the issue it mentions that it reduces boilerplate but I'm not seeing how this can reduce it.
> 
> But it got me thinking about a different API, instead of providing the token in the `forFeature` function, what about having a similar API to `USER_PROVIDED_META_REDUCERS`? This would look as follows:
> 
> ```ts
> providers: [
>   {
>     provide: USER_PROVIDED_EFFECTS,
>     useValue: [FooEffect, BarEffect]
>   }
> ]
> ```
> 
> The above only if it makes sense to have a similar feature...
> 
> If we implement this we should also add the feature to use a token in the `EffectsModule.forRoot()` function for consistency.

That would also work